### PR TITLE
dev-rokers: Add ROS

### PR DESCRIPTION
**Pull request by ludusrusso**

### CODE REVIEW

**Comment by ludusrusso** [comment]
@gmacario somethings I don't understand fails.. 

```
ERROR: Layer directory '/var/jenkins_home/workspace/s_genivi-dev-platform_PR-11-2HJQGHYLYNV3ROZBA6NZSIUGOHBWUCFWNSAT2DAFZS6RVUA3RWUQ/gdp-src-build/../openembedded-core/meta' does not exist! Please check BBLAYERS in /var/jenkins_home/workspace/s_genivi-dev-platform_PR-11-2HJQGHYLYNV3ROZBA6NZSIUGOHBWUCFWNSAT2DAFZS6RVUA3RWUQ/gdp-src-build/conf/bblayers.conf
```

But I think I correctly configured openembedded-core..

**Comment by gmacario** [comment]
Let me have a look

File: gdp-src-build/rokers-developer.conf [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 BBLAYERS += " \
   ${TOPDIR}/../meta-rokers \
+  ${TOPDIR}/../openembedded-core/meta \
+  ${TOPDIR}/../meta-openembedded/meta-oe \
+  ${TOPDIR}/../meta-openembedded/meta-python \
```
Comment (gmacario) [inline gdp-src-build/rokers-developer.conf]: Those three layers are not properly spelled, and anyway are already included by

* https://github.com/robotrokers/genivi-dev-platform/blob/dev-rokers/gdp-src-build/conf/templates/bblayers.inc#L9
* https://github.com/robotrokers/genivi-dev-platform/blob/dev-rokers/gdp-src-build/conf/templates/bblayers.inc#L14
* https://github.com/robotrokers/genivi-dev-platform/blob/dev-rokers/gdp-src-build/conf/templates/bblayers.inc#L18

I would just remove them from here

File: gdp-src-build/rokers-developer.conf [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 BBLAYERS += " \
   ${TOPDIR}/../meta-rokers \
```
Comment (gmacario) [inline gdp-src-build/rokers-developer.conf:4]: Please keep the entries in `BBLAYERS` sorted alphabetically to make code maintenance easier.

File: meta-rokers/recipes-rokers/images/rokers-image-base.bb [hunk 3]
```diff
@@ -30,6 +30,7 @@ PV ?= "snapshot+${DATE}"
 
 IMAGE_INSTALL_append = " \
     busybox-udhcpc \
+    roslaunch
```
Comment (gmacario) [inline meta-rokers/recipes-rokers/images/rokers-image-base.bb]: You forgot the trailing `\`!

File: .gitmodules [hunk 4]
```diff
@@ -47,3 +47,6 @@
 [submodule "meta-ros"]
 	path = meta-ros
 	url = https://github.com/bmwcarit/meta-ros
+[submodule "openembedded-core"]
+	path = openembedded-core
+	url = git://git.openembedded.org/openembedded-core
```
Comment (gmacario) [inline .gitmodules]: This commit is useless, the layer is already present as `meta-openembedded`.

See https://www.yoctoproject.org/tools-resources/projects/openembedded-core and https://stackoverflow.com/questions/35904769/what-are-the-differences-between-open-embedded-core-and-meta-openembedded

File: README.md [hunk 5]
```diff
@@ -55,3 +55,4 @@ To build:
     bitbake genivi-dev-platform
 
 Please see the README in meta-genivi-dev for further information.
+
```
Comment (gmacario) [inline README.md:58]: Remember to remove this commit from the final PR!

**Comment by ludusrusso** [comment]
@gmacario solved your comment, but still an error:

```
error "unsupported BN_BYTES"
```

:(

**Comment by gmacario** [comment]
@ludusrusso I have just submitted a new PR https://github.com/robotrokers/genivi-dev-platform/issues/12 which is supposed to fix and supersede this one.
Please have a look, then close this PR if you agree.

**Comment by ludusrusso** [comment]
Build fails on package `openssl-native`

See [build logs](https://build.rokers.io/job/robotrokers/job/genivi-dev-platform/view/change-requests/job/PR-11/22/console)

**Comment by gmacario** [comment]
Let us create another PR #15 to verify whether the build error is reproducible, or was just due to low disk space

**Comment by gmacario** [comment]
Build is now successful.

@ludusrusso Please test the generated image and report here

**Comment by ludusrusso** [comment]
I close the PR due to the jekyll issue! I'll create a new one!

**Pull request closed without merging**